src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java

```diff
@@ -0,0 +1,383 @@
+        }
+        super.neighborChanged(state, worldIn, pos, blockIn, fromPos);
+    }
+
+    @Override
+    public void breakBlock(final World worldIn, final BlockPos pos, final IBlockState state)
+    {
+        final TileEntity tileentity = worldIn.getTileEntity(pos);
+
+        if (tileentity instanceof TileEntityRack)
+        {
+            final IItemHandler handler = ((TileEntityRack) tileentity).getInventory();
+            InventoryUtils.dropItemHandler(handler, worldIn, pos.getX(), pos.getY(), pos.getZ());
+        }
+
+        super.breakBlock(worldIn, pos, state);
+    }
+
+    @NotNull
+    @Override
+    @SideOnly(Side.CLIENT)
+    public BlockRenderLayer getBlockLayer()
+    {
+        return BlockRenderLayer.SOLID;
+    }
+
+    @Override
+    public boolean onBlockActivated(
+                                     final World worldIn,
+                                     final BlockPos pos,
+                                     final IBlockState state,
+                                     final EntityPlayer playerIn,
+                                     final EnumHand hand,
+                                     final EnumFacing facing,
+                                     final float hitX,
+                                     final float hitY,
+                                     final float hitZ)
+    {
+        /*
+        If the world is client, open the gui of the building
+         */
+        if (!worldIn.isRemote)
+        {
+            final Colony colony = ColonyManager.getColony(worldIn, pos);
+            final TileEntity tileEntity = worldIn.getTileEntity(pos);
+
+            if ((colony == null || colony.getPermissions().hasPermission(playerIn, Action.ACCESS_HUTS))
+                  && tileEntity instanceof TileEntityRack)
+            {
+                playerIn.openGui(MineColonies.instance, 0, worldIn, pos.getX(), pos.getY(), pos.getZ());
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public void onBlockPlacedBy(
+                                 final World worldIn, final BlockPos pos, final IBlockState state, final EntityLivingBase placer, final ItemStack stack)
+    {
+        IBlockState tempState = state;
+        tempState = tempState.withProperty(VARIANT, RackType.byMetadata(stack.getItemDamage()));
+        tempState = tempState.withProperty(FACING, placer.getHorizontalFacing().getOpposite());
+
+        worldIn.setBlockState(pos, tempState, 2);
+    }
+
+    @Override
+    public void getSubBlocks(final Item itemIn, final CreativeTabs tab, final NonNullList<ItemStack> list)
+    {
+        list.add(new ItemStack(this, 1, RackType.DEFAULT.getMetadata()));
+    }
+
+    /**
+     * Convert the given metadata into a BlockState for this Block.
+     *
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+
+    @NotNull
+    @Override
+    @Deprecated
+    protected BlockStateContainer createBlockState()
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)